AWS-CREDENTIALS.md

```diff
@@ -4,7 +4,8 @@ Rusoto has the ability to source AWS access credentials in a few different ways:
 
 1. Environment variables via `rusoto::EnvironmentProvider` (`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`)
 2. AWS credentials file via `rusoto::ProfileProvider`
-3. IAM instance profile via `rusoto::IamProvider`
+3. IAM EC2 instance profile via `rusoto::InstanceMetadataProvider`
```

Comment:
I looked at the ordering official AWS SDKs use and it appears to be ECS container profiles then EC2 instance profiles.  The [Java SDK docs](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html) have the clearest outline of priority.  .NET and Ruby SDKs didn't mention container profiles at all in their pages on credential sourcing.